cpp/src/replace/clamp.cu

```diff
@@ -0,0 +1,385 @@
+
+template<typename T, typename ScalarIterator>
+std::unique_ptr<column> clamp(column_view const& input,
+                              ScalarIterator const& lo_itr,
+                              ScalarIterator const& lo_replace_itr,
+                              ScalarIterator const& hi_itr,
+                              ScalarIterator const& hi_replace_itr,
+                              rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                              cudaStream_t stream = 0) {
+    return clamper<T>(input, lo_itr, lo_replace_itr, hi_itr, hi_replace_itr, mr, stream);
+}
+
+struct dispatch_clamp {
+    template<typename T>
+    std::unique_ptr<column> operator ()(column_view const& input,
+                scalar const& lo,
+                scalar const& lo_replace,
+                scalar const& hi,
+                scalar const& hi_replace,
+                rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                cudaStream_t stream = 0) {
+        
+        auto lo_itr = make_pair_iterator<T>(lo);
+        auto hi_itr = make_pair_iterator<T>(hi);
+        auto lo_replace_itr = make_pair_iterator<T>(lo_replace);
+        auto hi_replace_itr = make_pair_iterator<T>(hi_replace);
+
+
+        return clamp<T>(input, lo_itr, lo_replace_itr,
+                     hi_itr, hi_replace_itr, mr, stream);
+    }
+};
+
+/**
+ * @copydoc cudf::experimental::clamp(column_view const& input,
+                                      scalar const& lo,
+                                      scalar const& lo_replace,
+                                      scalar const& hi,
+                                      scalar const& hi_replace,
+                                      rmm::mr::device_memory_resource* mr);
+ *
+ * @param[in] stream Optional stream on which to issue all memory allocations
+ */
+std::unique_ptr<column> clamp(column_view const& input,
+                              scalar const& lo,
+                              scalar const& lo_replace,
+                              scalar const& hi,
+                              scalar const& hi_replace,
+                              rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                              cudaStream_t stream = 0) {
+    CUDF_EXPECTS(lo.type() == hi.type(), "mismatching types of limit scalars");
+    CUDF_EXPECTS(lo_replace.type() == hi_replace.type(), "mismatching types of replace scalars");
+    CUDF_EXPECTS(lo.type() == lo_replace.type(), "mismatching types of limit and replace scalars");
+    CUDF_EXPECTS(lo.type() == input.type(), "mismatching types of scalar and input");
+
+    if ((not lo.is_valid(stream) and not hi.is_valid(stream)) or 
+        (input.is_empty())) {
+        // There will be no change
+        return std::make_unique<column>(input, stream, mr);
+    }
+
+    if (lo.is_valid(stream)) {
+        CUDF_EXPECTS(lo_replace.is_valid(stream), "lo_replace can't be null");
```

Comment:
Why not? I can't nullify all values less than a valid `lo`?

Reply:
I checked the requirement with @revans2 in #3794 which didn't have any such requirements. So adding that option will be little more expensive. So, I left it as future work if the requirement pops up.